### PR TITLE
feat: retry if server responds that

### PR DESCRIPTION
**Pull request by JoviDeCroock**
Fixes: https://github.com/apollographql/apollo-link/issues/995

- [ ] Make sure all of new logic is covered by tests and passes linting
- [ ] Update CHANGELOG.md with your change

This is a draft to see how everyone fares with this.

### CODE REVIEW

**Comment by JoviDeCroock** [comment]
CircleCI crashed....

File: packages/apollo-link-retry/src/retryLink.ts [hunk 1]
```diff
@@ -148,7 +148,21 @@ class RetryableOperation<TValue = any> {
...
     this.retryCount += 1;
-
+    // Retry if the server tells us to.
+    if (
+      error &&
+      error.response &&
+      error.response.status === 429 &&
+      error.response.headers['Retry-After']
+    ) {
+      this.scheduleRetry(
```
Comment (helfer) [inline packages/apollo-link-retry/src/retryLink.ts]: Doesn't this end up scheduling two retries if the response is a 429 (one in the `if` block, and one further down)?

**Comment by helfer** [comment]
Hm, now I see that the delay function doesn't get the error, it only gets the count. It might be worth passing the error as a second argument to delay. I think that would be pretty reasonable, because one might legitimately want to retry after a different amount of time for different errors.

**Comment by JoviDeCroock** [comment]
> Hm, now I see that the delay function doesn't get the error, it only gets the count. It might be worth passing the error as a second argument to delay. I think that would be pretty reasonable, because one might legitimately want to retry after a different amount of time for different errors.

We normally do pass it: https://github.com/apollographql/apollo-link/pull/1004/files#diff-cd9d084024219e2261760489591c11f2R174

Also given in typings of DelayFunction

**Pull request closed without merging**